src/bsoncxx/test/bson_builder.cpp

```diff
@@ -1669,69 +1754,207 @@ TEST_CASE("list builder appends inline nested", "[bsoncxx::builder::list]") {
     bson_append_array(&foo, "bar", -1, &bar);
     bson_append_document(&expected, "foo", -1, &foo);
 
-    builder::list b{"foo", {"bar", {1, 2, {"hello", "world"}}}};
```

Comment:
So, this type of usage isn't supported in this patch?  Because some of the `builder::list::document` constructors went away?

Reply:
Yes. You'd have to be explicit about the types of nested lists. Previously, we'd deduce it for the user.

Reply:
Acknowledged.